### ISSUE TITLE
Allow customizing Github release title

### ISSUE THREAD

**Issue by LucasPickering** [axodotdev/cargo-dist#631]
It'd be nice if you could customize the format of the title of each generated Github release. Currently, it looks like it pulls the title from the changelog if available, and uses the format `x.y.z - yyyy-mm-dd`. If there's no changelog, it just uses the git tag. I'm not 100% sure on that though, just my observation.

I'd like to customize that so the title is always in the format `vx.y.z`. I read through `announce.rs` a bit and didn't see an obvious place where this could be done, but I could take a crack at it with some direction.

**Comment by mistydemeo** [comment]
That's right, those are the places we get it from. We don't necessarily use the `.y.z - yyyy-mm-dd` format though; if we find a changelog entry matching the release version, we use that line whatever it is. For example, if we found a changelog line called "v0.6.0 - blah", we'd make a release with that title.

Just so I check what you need - is your goal to try to decouple the format from the changelog? Or is it just that you're interested in a specific release title, whatever route it takes to get there?

Thanks very much for offering to help out on this, by the way!

**Comment by LucasPickering** [comment]
@mistydemeo Yes, the goal is to decouple the title format from the changelog. I'm not sure exactly what the design would look like, but I'm imagining something similar to [Pre-release Replacements in cargo-release](https://github.com/crate-ci/cargo-release/blob/master/docs/reference.md#pre-release-replacements), where a format can be defined in `Cargo.toml`. Here's an example:

```toml
# Cargo.toml
[workspace.meta.dist]
release-title = "v{{version}}"
```

A few fields that I could imaging supporting here, based on the list in cargo-release:
- `version`
- `date`
- `crate_name`

As far as implementation, it looks like cargo-release uses a [very simple replacement technique](https://github.com/crate-ci/cargo-release/blob/3661ee59d6387cba152496aded1b922774254307/src/ops/replace.rs#L27), no templating libraries necessarily.
